test/theme_test.rb

```diff
@@ -3,49 +3,75 @@
 
 class ThemeTest < Minitest::Test
   def setup
-    @theme = make_theme(
-      "templates/index.liquid" => "",
-      "snippets/product.liquid" => "",
-      "sections/article-template/template.liquid" => "",
-      "locales/en.default.json" => "",
-    )
+    @themes = [
+      make_theme(
+        "templates/index.liquid" => "",
+        "snippets/product.liquid" => "",
+        "sections/article-template/template.liquid" => "",
+        "locales/en.default.json" => "",
+      ),
+      make_in_memory_theme(
+        "templates/index.liquid" => "",
+        "snippets/product.liquid" => "",
+        "sections/article-template/template.liquid" => "",
+        "locales/en.default.json" => "",
+      ),
+    ]
   end
 
   def test_all
-    assert_equal(4, @theme.all.size)
+    @themes.each do |theme|
+      assert_equal(4, theme.all.size)
+    end
   end
 
   def test_liquid
-    assert_equal(3, @theme.liquid.size)
+    @themes.each do |theme|
+      assert_equal(3, theme.liquid.size)
+    end
   end
 
   def test_json
-    assert_equal(1, @theme.json.size)
+    @themes.each do |theme|
+      assert_equal(1, theme.json.size)
+    end
   end
 
   def test_by_name
-    assert_equal("templates/index", @theme["templates/index"].name)
-    assert_equal("sections/article-template/template", @theme["sections/article-template/template"].name)
+    @themes.each do |theme|
+      assert_equal("templates/index", theme["templates/index"].name)
+      assert_equal("sections/article-template/template", theme["sections/article-template/template"].name)
+    end
   end
 
   def test_templates
-    assert_equal(["templates/index"], @theme.templates.map(&:name))
+    @themes.each do |theme|
+      assert_equal(["templates/index"], theme.templates.map(&:name))
+    end
   end
 
   def test_snippets
-    assert_equal(["snippets/product"], @theme.snippets.map(&:name))
+    @themes.each do |theme|
+      assert_equal(["snippets/product"], theme.snippets.map(&:name))
+    end
   end
 
   def test_sections
-    assert_equal(["sections/article-template/template"], @theme.sections.map(&:name))
+    @themes.each do |theme|
+      assert_equal(["sections/article-template/template"], theme.sections.map(&:name))
+    end
   end
 
   def test_default_locale_json
-    assert_equal(@theme["locales/en.default"], @theme.default_locale_json)
+    @themes.each do |theme|
+      assert_equal(theme["locales/en.default"], theme.default_locale_json)
+    end
   end
 
   def test_default_locale
-    assert_equal("en", @theme.default_locale)
+    @themes.each do |theme|
+      assert_equal("en", theme.default_locale)
+    end
   end
 
   def test_ignore
```

Comment:
I didn't port this test since I don't think it makes sense. This is strictly testing the constructor of Theme (not InMemoryTheme) and handles things that would be managed by the input files hash to InMemoryTheme.new.